iter/iter.go

```diff
@@ -5,9 +5,9 @@ import (
 	"github.com/ipld/go-ipld-prime"
 )
 
-// PBLinkItr behaves like an list of links iterator, even thought he HAMT behavior is more complicated
+// PBLinkItr behaves like an list of links iterator, even thought the HAMT behavior is more complicated
 type PBLinkItr interface {
```

Comment:
do we consider this contract public and therefore this would be a breaking change? istm that this should probably have been considered an internal detail, no?

Reply:
i don't think this is meant to be used externally, so agree that we should move to make it a private interface / not part of the external contract

Reply:
ack, have made it `pbLinkItr` now